webapp/src/components/PipelineSelect.tsx

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import { Select, MenuItem } from "@mui/material";
+
+type Props = {
+  selectedPipeline: number | string;
```

Comment:
I would let `PipelineSelect` deal with the `""` for no pipeline (an implementation detail), by supporting `undefined` and simply removing the two `?? ""` where `PipelineSelect` is called.
```suggestion
  selectedPipeline: number | undefined;
```